backend/apps/schemes/services/benefit_service.py

```
@@ -74,6 +74,17 @@ def benefit_create(*, benefit_data: dict, user=None):
                 "Benefit with this name and patient type already exists"
             )
 
+        # Handle plan field conversion
+        if 'plan' in benefit_data and benefit_data['plan']:
+            from apps.schemes.models import Plan
+            try:
+                plan = Plan.objects.get(id=benefit_data['plan'])
+                benefit_data['plan'] = plan
+            except Plan.DoesNotExist:
+                raise ValidationError(f"Plan with id {benefit_data['plan']} does not exist.")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)